<launch>
  <arg name="usb"/> 
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find magician_description)/urdf/magician.urdf.xacro'" />
  
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="50.0" />
    <param name="tf_prefix" type="string" value="" />
  </node>
    
  <node name="magician_hardware_node" pkg="magician_hardware" type="magician_hardware_node" args="$(arg usb)" output="screen"/>
  
  <rosparam file="$(find magician_hardware)/config/joint_state_controller.yaml" command="load"/>
  <node name="joint_state_controller_spawner" pkg="controller_manager" type="controller_manager" args="spawn joint_state_controller" respawn="false" output="screen"/>
  
  <rosparam file="$(find magician_hardware)/config/magician_arm_controller.yaml" command="load"/>
  <node name="elfin_arm_controller_spawner" pkg="controller_manager" type="controller_manager" args="spawn magician_arm_controller" respawn="false" output="screen"/>

</launch>
